src/rebar_utils.erl

```diff
@@ -673,3 +678,150 @@ cleanup_code_path(OrigPath) ->
         _ ->
             code:set_path(OrigPath)
     end.
+
+wordsize(Arch) when Arch =:= false; Arch =:= "" ->
+    native_wordsize();
+wordsize(Arch) ->
+    AllArchs = [
+                {"i686","32"},
+                {"i386","32"},
+                {"arm","32"},
+                {"aarch64", "64"},
+                {"x86_64","64"}
+               ],
+    case match_wordsize(Arch, AllArchs) of
+        false ->
+            case cross_wordsize(Arch) of
+                "" ->
+                    env_wordsize(os:getenv("REBAR_TARGET_ARCH_WORDSIZE"));
+                WordSize ->
+                    WordSize
+            end;
+        {_, Wordsize} ->
+            Wordsize
+    end.
+
+match_wordsize(Arch, [V={Match,_Bits}|Vs]) ->
+    case re:run(Arch, Match, [{capture, none}]) of
+        match ->
+            V;
+        nomatch ->
+            match_wordsize(Arch, Vs)
+    end;
+match_wordsize(_Arch, []) ->
+    false.
+
+env_wordsize(Wordsize) when Wordsize =:= false;
+                            Wordsize =:= "" ->
+    ?WARN("REBAR_TARGET_ARCH_WORDSIZE not set, assuming 32\n", []),
+    "32";
+env_wordsize(Wordsize) ->
+    try list_to_integer(Wordsize) of
+        16 -> "16";
+        32 -> "32";
+        64 -> "64";
+        _ ->
```

Comment:
Nice catch, I don't know why @tonyrog checks if it's an integer. Thanks, simplified.